components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/LoginContextManagementUtil.java

```diff
@@ -323,14 +325,42 @@ private static boolean isStepHasMultiOption(AuthenticationContext context) {
         return stepHasMultiOption;
     }
 
-    private static boolean isValidRequest(HttpServletRequest request, AuthenticationContext context) {
+    private static boolean canHandleAuthenticator(AuthenticationContext context, String authenticators) {
+
+        List<String> authenticatorsList = new ArrayList();
+        if (authenticators != null) {
+            String[] authenticatorIdPMappings = authenticators.split(";");
+            for (String authenticatorIdPMapping : authenticatorIdPMappings) {
+                String[] authenticatorIdPMapArr = authenticatorIdPMapping.split(":");
+                authenticatorsList.add(authenticatorIdPMapArr[0]);
+            }
+        }
+
+        Map<Integer, StepConfig> stepMap = context.getSequenceConfig().getStepMap();
+        if (stepMap != null && !stepMap.isEmpty()) {
+            StepConfig stepConfig = stepMap.get(context.getCurrentStep());
+
+            for (AuthenticatorConfig authenticatorConfig : stepConfig.getAuthenticatorList()) {
+                if (authenticatorsList.contains(authenticatorConfig.getName())) {
+                    return true;
+                }
+            }
+        }
+        return false;
```

Comment:
When the same authenticator is there in multiple steps, this method fails. An example scenario that will fail with this fix would be as follows.

- Configure basic authentication and emailOTP in the first step, configure emailOTP and SMSOTP in the second step.
- Login through basic and stop at the multi options page.
- Return back to the login page.
- Try to login again.
- An error will be thrown in the UI saying "Something went wrong during the authentication process. Please try signing in again.".

A suggestion to fix the issue would be as follows.

```suggestion
            for (AuthenticatorConfig authenticatorConfig : stepConfig.getAuthenticatorList()) {
                if (!authenticatorsList.contains(authenticatorConfig.getName())) {
                    return false;
                }
            }
            return true;
        }
        return false;
```